src/LinuxTracing/LibunwindstackMultipleOfflineAndProcessMemoryTest.cpp

```diff
@@ -0,0 +1,170 @@
+  StackSliceView stack_slice(kStartAddress, bytes.size(), bytes.data());
+
+  std::shared_ptr<unwindstack::Memory> sut =
+      LibunwindstackMultipleOfflineAndProcessMemory::CreateWithoutProcessMemory({stack_slice});
+
+  std::array<char, 3> destination{};
+  destination.fill(0x11);
+  size_t read_count = sut->Read(kStartAddress - 1, destination.data(), 3);
+
+  ASSERT_EQ(read_count, 0);
+  EXPECT_EQ(destination[0], 0x11);
+  EXPECT_EQ(destination[1], 0x11);
+  EXPECT_EQ(destination[2], 0x11);
+}
+
+TEST(LibunwindstackMultipleOfflineAndProcessMemory,
+     RequestingToReadUnknownMemoryWithoutProcessReturnsZero) {
+  constexpr uint64_t kStartAddress = 0xADD8E55;
+  std::vector<char> bytes{0x01, 0x10, 0x20, 0x30, 0x40};
+  StackSliceView stack_slice(kStartAddress, bytes.size(), bytes.data());
+
+  std::shared_ptr<unwindstack::Memory> sut =
+      LibunwindstackMultipleOfflineAndProcessMemory::CreateWithoutProcessMemory({stack_slice});
+
+  std::array<char, 3> destination{};
+  destination.fill(0x11);
+  size_t read_count = sut->Read(0xFE, destination.data(), 3);
+
+  ASSERT_EQ(read_count, 0);
+  EXPECT_EQ(destination[0], 0x11);
+  EXPECT_EQ(destination[1], 0x11);
+  EXPECT_EQ(destination[2], 0x11);
+}
+
+TEST(LibunwindstackMultipleOfflineAndProcessMemory,
+     ReadFromCompleteMemoryEvenIfPartiallyIntersectsWithOtherStackSlice) {
+  constexpr uint64_t kStartAddress1 = 0xADD8E55;
+  std::vector<char> bytes1{0x01, 0x10, 0x20, 0x30, 0x40};
+  StackSliceView stack_slice1(kStartAddress1, bytes1.size(), bytes1.data());
+
+  constexpr uint64_t kStartAddress2 = kStartAddress1 - 2;
+  std::vector<char> bytes2{0x0a, 0x0b, 0x0c, 0x0d, 0x0f, 0x10, 0x01};
+  StackSliceView stack_slice2(kStartAddress2, bytes2.size(), bytes2.data());
+
+  std::shared_ptr<unwindstack::Memory> sut =
+      LibunwindstackMultipleOfflineAndProcessMemory::CreateWithoutProcessMemory(
+          {stack_slice1, stack_slice2});
+
+  std::array<char, 3> destination{};
+  size_t read_count = sut->Read(kStartAddress2, destination.data(), 3);
+
+  ASSERT_EQ(read_count, 3);
+  EXPECT_EQ(destination[0], 0x0a);
+  EXPECT_EQ(destination[1], 0x0b);
+  EXPECT_EQ(destination[2], 0x0c);
+}
+
+TEST(LibunwindstackMultipleOfflineAndProcessMemory, ReadFromTestProcess) {
+  constexpr uint64_t kStartAddress1 = 0xADD8E55;
+  std::vector<char> bytes1{0x01, 0x10, 0x20, 0x30, 0x40};
+  StackSliceView stack_slice1(kStartAddress1, bytes1.size(), bytes1.data());
+
+  constexpr uint64_t kStartAddress2 = 0xABCDEF;
+  std::vector<char> bytes2{0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17};
+  StackSliceView stack_slice2(kStartAddress2, bytes2.size(), bytes2.data());
+
+  std::vector<StackSliceView> stack_slices{stack_slice1, stack_slice2};
+  std::vector<LibunwindstackOfflineMemory> stack_memories{};
+  stack_memories.reserve(stack_slices.size());
+  for (const StackSliceView& stack_slice_view : stack_slices) {
+    stack_memories.emplace_back(stack_slice_view);
+  }
```

Comment:
`stack_slices` and `stack_memories` are now unused so these lines can be removed.

Reply:
Right. Removed it.